### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by ammar92**
### Changes
This updates and removes some stale or insecure dependencies. Pre-commit hooks have also been updated and fixed accordingly.

Closes #2367
Closes #2368 

There are still some major package upgrades that need to be done for some of the modules, such as `sqlalchemy` 2, but this requires a bit more attention because migration files need to be manually upgraded.

---

### Code Checklist
- [x] All the commits in this PR are properly PGP-signed and verified.
- [ ] This PR only contains functionality relevant to the issue; tickets have been created for newly discovered issues.
- [ ] I have written unit tests for the changes or fixes I made.
- [ ] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

### Communication
- [ ] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [ ] I have made corresponding changes to the documentation, if necessary.
- [ ] I have included comments in the code to elaborate on what is not self-evident from the code itself, including references to issues and discussions online, or implicit behavior of an interface.

---
## Checklist for code reviewers:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---
## Checklist for QA:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

**Comment by ammar92** [comment]
Have been clicking around and it doesn't seem to not break anything

File: rocky/katalogus/templates/katalogus_settings.html [hunk 1]
```diff
@@ -44,13 +44,11 @@ <h1>{% translate "KAT-alogus Settings" %}</h1>
                     </div>
                 </div>
```
Comment (dekkers) [inline rocky/katalogus/templates/katalogus_settings.html:45]: This div inside the else clause needs to be removed, not the one outside. If this is fixed the endif should also be indented on the same level as the if/else.

Comment (ammar92) [reply rocky/katalogus/templates/katalogus_settings.html:45]: Good catch. Will fix this

**Comment by stephanie0x00** [comment]
## Checklist for QA:
- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
    - [x] I have managed to pass the onboarding flow
    - [x] Objects and Findings are created properly
    - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.

### What works:
All seems to work.

### What doesn't work:
n/a

### Bug or feature?:
n/a

**Review by dekkers — APPROVED** [review]

**Pull request merged**